src/js/client/templateFormControl.js

```diff
@@ -0,0 +1,222 @@
+/*
+
+  A component to standardize handling of "simple" form data submitted via AJAX.  This component:
+
+  1.  Performs an initial render of the component using the template specified in `options.templates.initial`
+  2.  Binds a form submission that sends the `model` data using the options specified in `options.ajax`
+  3.  If the AJAX request returns an error or the result contains a "falsy" `ok` variable:
+      a)  Any previous "success" messages are cleared.
+      b)  the error is transformed using the rules found in `options.rules.error`
+      c)  the results are passed along to the `error` subcomponent for display.
+  4.  If the AJAX request is successful:
+      a)  Any previous "error" messages are cleared.
+      b)  The results are transformed using the rules found in `options.rules.success` and passed to the `success` subcomponent for display.
+      c)  The results are transformed using the rules found in `options.rules.model`, and applied to the model using the change applier.
+
+ */
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.hb.client.templateFormControl");
+
+
+    gpii.templates.hb.client.templateFormControl.checkRequirements = function (that) {
+        var errors = [];
+
+        if (!that.options.templates || !that.options.templates.initial || !that.options.templates.error || !that.options.templates.success) {
```

Comment:
Cool. I think while we're waiting for the schema handling (which might be a large piece of intricate work, and may never provide enormously readable diagnostics), a simple 3-line function fitting the above contract would greatly improve the experience of users supplying options to gpii-handlebars. We can certainly come back and revisit it once the schema system has been implemented.
As reference, the worthy HAVERBEKE recently tweeted asking for pointers to JSON validation approaches and here are the responses he got:
https://twitter.com/marijnjh/status/606395218066309120